include/deal.II/numerics/vector_tools.h

```diff
@@ -924,6 +930,64 @@ namespace VectorTools
                 VectorType &vec_result);
 
   /**
+   * The same as above for parallel Triangulations.
+   * If you use more than four components or the degree of the used
+   * FiniteElement is higher than eight, use project_parallel_generic() instead.
+   *
+   * Currently, the implementation only allows for @p enforce_zero_boundary
+   * and @p project_to_boundary_first to be <tt>false</tt>.
+   * @p vec_result is assumed to not have a ghost elements.
+   */
+  template <int dim, typename VectorType, int spacedim>
+  void project_parallel (const Mapping<dim, spacedim>   &mapping,
```

Comment:
i think the idea of @bangerth and @kronbichler is not to have any `project_parallel` in `vector_tools.h`. You only add them in internal namespace of `vector_tools.templates.h` to do the switching.